### PR TITLE
Unify Blake2bHash and Digest (take 2)

### LINKED ISSUE #2008: Unify `Digest` and `Blake2bHash` Types
The `Digest` and `Blake2bHash` types duplicate functionality and should be unified.

### PR DESCRIPTION
**Pull request by dwerner**
Takes the place of https://github.com/casper-network/casper-node/pull/2009

Resolves #2008

This PR unifies the Digest and Blake2bHash.

### CODE REVIEW

File: execution_engine/Cargo.toml [hunk 1]
```diff
@@ -15,6 +15,7 @@ anyhow = "1.0.33"
 base16 = "0.2.1"
 bincode = "1.3.1"
 blake2 = "0.9.0"
```
Comment (xcthulhu) [inline execution_engine/Cargo.toml]: Probably should remove this...

Comment (dwerner) [reply execution_engine/Cargo.toml]: Ah good catch.

**Review by Fraser999 — APPROVED** [review]

**Pull request merged**